src/main/scala/mesosphere/marathon/api/v2/validation/AppValidation.scala

```diff
@@ -304,7 +305,7 @@ trait AppValidation {
     }
   )
 
-  def validateCanonicalAppUpdateAPI(enabledFeatures: Set[String]): Validator[AppUpdate] = forAll(
+  def validateCanonicalAppUpdateAPI(enabledFeatures: Set[String], normalizationConfig: AppNormalization.Config): Validator[AppUpdate] = forAll(
```

Comment:
We already had an discussion regarding this issue. There is an explicit override 
```
override def defaultNetworkName: Option[String] =
  | env.vars.get(DefaultNetworkNameForMigratedApps).orElse(networkName).orElse(throw SerializationFailedException(
  | MigrationFailedMissingNetworkEnvVar))


```
 which throws an exception if this is is None. But None is a valid state in some cases afterwards. Therefore we decided to pass this not directly and go for a lazy implementation. 

What about adding a trait `ValidationConfig` which holds the `enabledFeatures` and `defaultNetworkName` ?